src/include/distributed/log_utils.h

```diff
@@ -9,7 +9,22 @@
 #ifndef LOG_UTILS_H
 #define LOG_UTILS_H
 
+
+#include "utils/guc.h"
+
+
 extern bool IsLoggableLevel(int logLevel);
-extern char * ApplyLogRedaction(const char *text);
+extern char * HashLogMessage(const char *text);
+
+#define ApplyLogRedaction(text) \
+	(log_min_messages <= ereport_loglevel ? HashLogMessage(text) : text)
+
+#undef ereport
+#define ereport(elevel, rest) \
+	do { \
+		int ereport_loglevel = elevel; \
```

Comment:
maybe `int ereport_loglevel = (elevel)` & then `ereport_domain(ereport_loglevel, TEXTDOMAIN, rest)` which also removes need for void cast to avoid usage warnings

Reply:
We need elevel to be a compile time constant for the macro to inject pg_unreachable. Otherwise we get a bunch of compile errors.